main/src/cgeo/geocaching/utils/LogWriter.java

```diff
@@ -0,0 +1,71 @@
+package cgeo.geocaching.utils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+
+public class LogWriter {
+    private PrintWriter logWriter = null;
+    private String prefix;
+
+    public LogWriter(final String prefix) {
+        this.prefix = prefix;
+        checkLogfile();
+    }
+
+    public void log(final String info) {
+        checkLogfile();
+        if (null != logWriter) {
+            logWriter.println("--------------------------------------------------------------------");
+            logWriter.println(CalendarUtils.formatDateTime("yyyy-MM-dd HH:mm:ss.SSS"));
+            logWriter.println(info);
+        }
+    }
+
+    public void d(final String info) {
+        Log.d(info);
```

Comment:
My understand is that LogWriter is for logs too long to be handled by logcat? In this case, why log the info additionally to logcat (it won't be able to handle it anyway). Instead, maybe a link to the LogWriter file should be logged.

Reply:
Strings which may be too long for logcat should probably use `logWriter.log()` instead of the `.i` etc. variants. But even if not, the full string would be logged to the individual log file, whereas the string in logcat would be truncated.

Only those strings logged using one of the `logWriter.i()`, `logWriter.e()` etc. variants also log to the regular logcat. This is to have logcat unchanged compared to not using `logWriter`, and you can additionally turn on the separate logfile, if needed.

Reply:
Ok understood